crates/sovereign-sdk/module-system/sov-modules-rollup-blueprint/src/lib.rs

```diff
@@ -57,20 +56,18 @@ pub trait RollupBlueprint: Sized + Send + Sync {
     type NativeRuntime: RuntimeTrait<Self::NativeContext, Self::DaSpec> + Default + Send + Sync;
 
     /// Prover service.
-    type ProverService: ProverService<
-            Self::Vm,
-            StateRoot = <<Self::NativeContext as Spec>::Storage as Storage>::Root,
-            Witness = <<Self::NativeContext as Spec>::Storage as Storage>::Witness,
-            DaService = Self::DaService,
-        > + Send
-        + Sync
-        + 'static;
+    type ProverService: ProverService<Self::Vm, DaService = Self::DaService> + Send + Sync + 'static;
 
     /// Creates a new instance of the blueprint.
     fn new() -> Self;
 
-    /// Get code commitments by fork.
-    fn get_code_commitments_by_spec(&self) -> HashMap<SpecId, <Self::Vm as Zkvm>::CodeCommitment>;
+    /// Get batch prover code commitments by fork.
+    fn get_batch_prover_code_commitments_by_spec(
+        &self,
+    ) -> HashMap<SpecId, <Self::Vm as Zkvm>::CodeCommitment>;
+
+    /// Get light client prover code commitment.
+    fn get_light_client_prover_code_commitment(&self) -> <Self::Vm as Zkvm>::CodeCommitment;
```

Comment:
Why get_batch_prover_code_commitments_by_spec returns HashMap but this does not?

Reply:
well we can change this later easily